Updated optimal-edge-discovery.yaml with latest guidelines

urvika-v opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* enhancement/feature



#### What this PR does / why we need it:




#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #

#### Special notes for reviewers:



#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.



```
docs

```

jgarciatovar commented:
> @jgarciatovar Thank you for the comments. I have removed 405 and 5xx errors from the API spec. Updated x-correlator and added DeviceResponse. Please review it .

The `IDENTIFIER_MISSMATCH` error should be removed after the addition of the `DeviceResponse`. You can check updates on SED or AED for guidance.

While SED response consists of a single Zone returned, OED might return multiple zones, so you have set an array of `EdgeCloudZone` field. However, there will be a unique `ApplicationProfileId` and a unique `DeviceResponse` for all the zones returned. Thus, it seems better approach to me that the response is a single new object that includes single `DeviceResponse`, single `ApplicationProfileId`, and array of `EdgeCloudZone` zones.

jgarciatovar commented on code/API_definitions/optimal-edge-discovery.yaml:
```diff
@@ -0,0 +1,810 @@
+    application endpoints.
+
+    These APIs may be called either by an application client hosted on a device
+    attached to the operator network (i.e. phone, tablet) for following
```
You need to add the option to trigger this API by the AP server:

``` 
(i.e. phone, tablet), *or by the server* for the following scenario:
```

jgarciatovar commented on code/API_definitions/optimal-edge-discovery.yaml:
```diff
@@ -0,0 +1,810 @@
+        zones to deploy application to minimize latency for users in a specific
+        region. This can be achieved using the
+        `/retrieve-optimal-edge-cloud-zones` API.
+
```
Missing `GET /regions` path in the documentation

urvika-v replied:
Updated

jgarciatovar commented on code/API_definitions/optimal-edge-discovery.yaml:
```diff
@@ -0,0 +1,810 @@
+
+    # Responses
+
+    ## Success
```
The success response in the documentation has not been updated to the latest changes.

urvika-v replied:
Updated

jgarciatovar commented on code/API_definitions/optimal-edge-discovery.yaml:
```diff
@@ -0,0 +1,810 @@
+  - url: "{apiRoot}/optimal-edge-discovery/vwip"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
```
Shouldn't be https? I think it needs to be updated to https, but not 100% sure. @Kevsy can you comment?

urvika-v commented:
> @urvika-v - I've added some comments after the last updates. Other than those comments, LGTM

@jgarciatovar - Thank you for the comments. I have updated the spec. Could you please review.

jgarciatovar commented on code/API_definitions/optimal-edge-discovery.yaml:
```diff
@@ -0,0 +1,840 @@
+                title: EdgeCloudZonesResponse
+                type: array
+                items:
+                  $ref: "#/components/schemas/EdgeDiscoveryResponse"
```
Given the last changes in the schemas, do you need this element to be an array? Or is this a single element which might contain an array of zones inside? Is there any use case that requires this element to be an array of `EdgeDiscoveryResponse` objects?

Kevsy reviewed (approved):
LGTM
